pq-crypto/sike_r1/sike_r1_kem.c

```diff
@@ -117,9 +117,9 @@ int SIKE_P503_r1_crypto_kem_dec(unsigned char *ss, const unsigned char *ct, cons
 
     // Generate shared secret ss <- H(m||ct) or output ss <- H(s||ct)
     EphemeralKeyGeneration_A(ephemeralsk_.d, c0_);
-    if (memcmp(c0_, ct, SIKE_P503_R1_PUBLIC_KEY_BYTES) != 0) {
-        memcpy(temp, sk, MSG_BYTES);
-    }
+    bool copy_or_dont = s2n_constant_time_equals(c0_, ct, SIKE_P503_R1_PUBLIC_KEY_BYTES);
+    s2n_constant_time_copy_or_dont(temp, sk, MSG_BYTES, copy_or_dont);
```

Comment:
We might also want to note in a comment that this is an intentional deviation from the NIST version. Our [readme](https://github.com/awslabs/s2n/blob/36f6962b5f7e3823d9acac385f3879f424f44e1f/pq-crypto/sike_r1/Readme.md) for this code says that "only minor modifications were made in order to integrate this code into s2n's build system", and I don't think this change fits that.